### ISSUE TITLE
Conda environment with singularity container and conda+singularity prefix

### ISSUE THREAD

**Issue by leachim** [snakemake/snakemake#179]
**Snakemake version**
5.9.1

**Describe the bug**
It's not possible to run conda with a conda environment in a singularity image, when specifying both conda and singularity prefixes. 

**Logs**
`snakemake -s Snakefile --use-conda --conda-prefix=/home/$USER/tmp/Conda --use-singularity --singularity-prefix=/home/$USER/tmp/Singularity --singularity-args "--bind /home/$USER/tmp:/home/$USER/tmp"`
If applicable, any terminal output to help explain your problem.
```
Building DAG of jobs...
Creating conda environment envs/environment.yml...
Downloading and installing remote packages.
CreateCondaEnvironmentException:
Could not create conda environment from /home/$USER/tutorial/envs/environment.yml:

EnvironmentFileNotFound: '/home/$USER/tmp/Conda/bba6ce32.yaml' file not found


  File "/home/$USER/.anaconda3/lib/python3.7/site-packages/snakemake/deployment/conda.py", line 314, in create
```

**Minimal example**
Using the tutorial (v5.4.5), with this modification to the first rule of the bam file: 
    singularity:                                                                                                                                                            
          "docker://continuumio/miniconda:4.7.12"                                                                                                                             
      conda:                                                                                                                                                                  
          "envs/environment.yml" 

where envs/environment.yml is the yaml file from the tutorial.

**Comment by ManavalanG** [comment]
Based on my experience, snakemake doesn't resolve `$USER`, which causes the problem. For debugging, replace `$USER` with its value and then give it a try.

**Comment by leachim** [comment]
No, that is not the issue. It is actually resolved to my username and I also tried to use it with my real username. The error is the same.

**Comment by leachim** [comment]
Might have something to do with the accessibility of the conda temporary yaml file in the container?

**Comment by alpha-beta-soup** [comment]
I'm seemingly getting the same issue with Snakemake v6.2.1

**Comment by m-jahn** [comment]
Closing this because the interaction of snakemake with containers has significantly changed now.
There is the possibility to use `--containerize` to build a custom container with all conda envs, or to supply global or rule-based custom containers with arbitrary envs.
